### PR TITLE
Docker and K8s: Fix shell arguments when split by the runner

### PR DESCRIPTION
**Pull request by nikola-jokic**
When runner passes arguments to the hook, there may be situations where quoted strings are interpreted as separate arguments.

This PR addresses the issue and fixes the arguments based on `shlex` expansion

Fixes the [issue in this discussion](https://github.com/orgs/community/discussions/70488)

### CODE REVIEW

File: packages/docker/src/utils.ts [hunk 1]
```diff
@@ -84,6 +86,10 @@ export function sanitize(val: string): string {
   return newNameBuilder.join('')
 }
 
+export function fixArgs(args: string[]): string[] {
+  return shlex.split(args.join(' '))
```
Comment (fhammerl) [inline packages/docker/src/utils.ts:90]: Is `powershell` ok with this?

Comment (nikola-jokic) [reply packages/docker/src/utils.ts:90]: If arguments are not POSIX complaint, adding quotes around them would fix the problem

**Comment by timmjd** [comment]
@nikola-jokic : does this cover the same as described in #117?

**Comment by timmjd** [comment]
Awesome!

This implementation is even better than #117 due to you keep the array consistency.

### Example 1
```yaml
name: Build MKDocs

runs:
  using: docker
  image: docker://squidfunk/mkdocs-material:latest
  entrypoint: mkdocs
  args:
  - build --verbose --strict
```

```yaml
spec:
  containers:
  - command:
    - mkdocs
    args:
    - build --verbose --strict
```

### Example 2
```yaml
name: Build MKDocs

runs:
  using: docker
  image: docker://squidfunk/mkdocs-material:latest
  entrypoint: mkdocs
  args:
  - build
  - --verbose
  - --strict
```

```yaml
spec:
  containers:
  - command:
    - mkdocs
    args:
    - build
    - --verbose
    - --strict
```

**Review by fhammerl — APPROVED** [review]
LGTM

**Pull request merged**